### PR TITLE
Set default auto-instrumentation java image

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <p.loffay@gmail.com>

This PRs set default auto-instrumentation java image if the image is not specified in the CR. The operator does the same for OTELcol and allocator images.

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -1,2 +1,8 @@
 resources:
-  - manager.yaml
+- manager.yaml
```
Comment (VineethReddy02) [inline config/manager/kustomization.yaml:2]: These changes shouldn't be committed. AFAIK they get changes during the local build, deployment.

**Comment by pavolloffay** [comment]
This is ready for review

**Review by VineethReddy02 — APPROVED** [review]
LGTM!
@jpkrohling can you take a final look?

File: pkg/instrumentation/podmutator.go [hunk 2]
```diff
@@ -111,3 +114,10 @@ func (pm *instPodMutator) selectInstrumentationInstanceFromNamespace(ctx context
 		return otelInsts.Items[0], nil
 	}
 }
+
+func (pm *instPodMutator) applyDefaults(otelinst v1alpha1.Instrumentation) v1alpha1.Instrumentation {
+	if otelinst.Spec.Java.Image == "" {
```
Comment (jpkrohling) [inline pkg/instrumentation/podmutator.go:119]: This should be part of the Defaulter.

https://github.com/open-telemetry/opentelemetry-operator/blob/de6219f576b1af2e88c9119452ae8ef1db67cb7a/apis/collector/v1alpha1/opentelemetrycollector_webhook.go#L39-L55

Comment (pavolloffay) [reply pkg/instrumentation/podmutator.go:119]: Collector image is not handled in the defaulter either. 

That said I might agree with you (I need to test the defaulter, when it comes to updates).  I have this WIP PR https://github.com/open-telemetry/opentelemetry-operator/pull/517 that might change this to use the defaulter.

Comment (jpkrohling) [reply pkg/instrumentation/podmutator.go:119]: The collector image was probably handled before we implemented the defaulter.

Comment (pavolloffay) [reply pkg/instrumentation/podmutator.go:119]: I would like to get this merged and then change just remove this function in the #517 PR.

**Pull request merged**